tokio/src/sync/watch.rs

```diff
@@ -574,20 +574,88 @@ impl<T> Sender<T> {
     pub fn send_modify<F>(&self, func: F)
     where
         F: FnOnce(&mut T),
+    {
+        self.send_if_modified(|value| {
+            func(value);
+            true
+        });
+    }
+
+    /// Modifies watched value, notifying all receivers if modified.
+    ///
+    /// This can useful for modifying the watched value, without
+    /// having to allocate a new instance. Additionally, this
+    /// method permits sending values even when there are no receivers.
+    ///
+    /// The closure that modifies the value must return `true` if the
+    /// value has actually been modified. It should only return `false`
+    /// if the value is guaranteed to be unnmodified despite the mutable
+    /// borrow. Receivers are only notified if the value has been modified.
```

Comment:
While we are at it: Should we remove the redundancy in the docs and refer to the more versatile `send_if_modified()` in the docs for `send_modify()`? The docs would then basically reflect the forwarding of the implementation.

Reply:
Any improvements to the docs you can come up with would be good!

Reply:
I have revised and aligned the docs of both functions, highlighting their differences and similarities. Since I am not a native speaker someone should verify.

Removing the redundancies would be difficult. Copying the shared parts should be ok to keep the descriptions readable.